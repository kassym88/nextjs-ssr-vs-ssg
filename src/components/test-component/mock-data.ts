import { LAUNCHES_QUERY } from '../../queries';

export const variables = {
  limit: 10,
  sort: 'mission_name',
};

export const mocks = [
  {
    request: {
      query: LAUNCHES_QUERY,
      variables,
    },
    result: {
      data: {
        launchesPast: [
          {
            id: 1,
            mission_name: 'test-name-1',
            launch_date_local: '2020-10-24T11:31:00-04:00',
            links: {
              video_link: 'test-link-1',
            },
          },
          {
            id: 2,
            mission_name: 'test-name-2',
            launch_date_local: '2020-10-24T11:31:00-04:00',
            links: {
              video_link: 'test-link-2',
            },
          },
        ],
      },
    },
  },
];