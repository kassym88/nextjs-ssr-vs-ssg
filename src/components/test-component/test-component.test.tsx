import { render, act } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';

import TestComponent from './test-component';
import { mocks, variables as mockVariables } from './mock-data';

jest.mock('next/router', () => ({
  useRouter: () => ({
      route: '/',
      pathname: '',
      query: mockVariables,
      asPath: '',
    }),
}));

describe('TestComponent', () => {
  it('should render correctly', async () => {
    const wrapper = render(
      <MockedProvider
        mocks={mocks}
        addTypename={false}
        // MockedProvider not working as described in documentation: data from useQuery hook always undefined
        // https://github.com/apollographql/react-apollo/issues/1747#issuecomment-531752539
        defaultOptions={{
          watchQuery: {
            fetchPolicy: 'network-only',
          }}}
      >
        <TestComponent label="Test" />
      </MockedProvider>
    );

    await act(async () => {
      await new Promise(setImmediate);
    });

    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
