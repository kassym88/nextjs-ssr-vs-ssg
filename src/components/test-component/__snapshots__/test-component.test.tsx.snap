// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestComponent should render correctly 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Test
    </h1>
    <div>
      <a
        href="test-link-1"
      >
        <h3>
          test-name-1
        </h3>
        <p>
          <strong>
            Launch Date: 
          </strong>
          10/24/2020
        </p>
      </a>
      <a
        href="test-link-2"
      >
        <h3>
          test-name-2
        </h3>
        <p>
          <strong>
            Launch Date: 
          </strong>
          10/24/2020
        </p>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
